### PR TITLE
Make internal implementation details private

### PR DESCRIPTION
**Pull request by sosthene-nitrokey**
This will allow us to progressively evolve the implementation without having breaking changes everywhere.

### CODE REVIEW

**Comment by robin-nitrokey** [comment]
Please rebase onto main.

**Comment by sosthene-nitrokey** [comment]
Just done

File: src/constants.rs [hunk 1]
```diff
@@ -10,12 +10,14 @@ pub const CLASS_CCID: u8 = 0x0B;
 pub const SUBCLASS_NONE: u8 = 0x0;
 
 #[repr(u8)]
-pub enum TransferMode {
-    // bulk transfers, optional interrupt IN
+pub(crate) enum TransferMode {
```
Comment (robin-nitrokey) [inline src/constants.rs:13]: The module is no longer public so I think this change is not needed.

File: src/lib.rs [hunk 2]
```diff
@@ -13,8 +13,8 @@ extern crate delog;
 generate_macros!();
 
 pub mod class;
```
Comment (robin-nitrokey) [inline src/lib.rs:15]: Can we private too.

File: src/lib.rs [hunk 3]
```diff
@@ -13,8 +13,8 @@ extern crate delog;
...
-pub mod constants;
-pub mod pipe;
+mod constants;
+mod pipe;
 pub mod types;
```
Comment (robin-nitrokey) [inline src/lib.rs:18]: Remove `pub` here and just re-export `types::Status`?

**Review by robin-nitrokey — CHANGES_REQUESTED** [review]

**Review by robin-nitrokey — APPROVED** [review]

**Pull request merged**